### PR TITLE
Fix dropping eds cache

### PR DESCRIPTION
**Pull request by ilackarms**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2356

### CODE REVIEW

**Comment by yuval-k** [comment]
ci failed on code format
```

index 1cc5ca0d..895fd75e 100644
--- a/go.mod
+++ b/go.mod
@@ -47,6 +47,7 @@ require (
 	github.com/onsi/gomega v1.8.1
 	github.com/opencontainers/go-digest v1.0.0-rc1
 	github.com/pkg/browser v0.0.0-20180916011732-0a3d74bf9ce4
+	github.com/pkg/errors v0.8.1
 	github.com/prometheus/client_golang v1.2.1
 	github.com/prometheus/client_model v0.0.0-20190812154241-14fe0d1b01d4
 	github.com/prometheus/prometheus v2.5.0+incompatible
```

File: projects/gloo/pkg/discovery/discovery_test.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		// update eds by sending another eds list
+
+		select {
+		case redChan <- v1.EndpointList{ep("red-2"), ep("red-3")}: // change to 2 red endpoints
+		default:
+			Fail("expected redChan to be empty")
+		}
+
+		// expect 1 blue always
+		Consistently(listEps).Should(ContainElement("endpoint-blue-1"))
```
Comment (CamelCaseNotation) [inline projects/gloo/pkg/discovery/discovery_test.go]: https://www.youtube.com/watch?v=ElJe5M54brI

**Comment by CamelCaseNotation** [comment]
Is there anyway I can look at the failed build, or is that only for solo employees?

**Comment by ilackarms** [comment]
@CamelCaseNotation our CI bot used to publish the build logs at a public endpoint, but the behavior has been flaky recently.

**Comment by ilackarms** [comment]
/kick

@CamelCaseNotation  we have a high number of flakes right now, the test failure was due to a flake in one of the local e2e tests

this was the ci failure:
```


Step #11 - "test": 
Step #11 - "test": ------------------------------
Step #11 - "test": â€¢ Failure [11.026 seconds]
Step #11 - "test": Gateway in memory traffic [It] should work with no ssl and cleans up the envoy config when the virtual service is deleted 
Step #11 - "test": /workspace/gloo/test/e2e/gateway_test.go:210
Step #11 - "test": 
Step #11 - "test":   Timed out after 10.000s.
Step #11 - "test":   Expected
Step #11 - "test":       <bool>: false
Step #11 - "test":   to be true
Step #11 - "test": 
Step #11 - "test":   /workspace/gloo/test/e2e/gateway_test.go:230
Step #11 - "test": 
Step #11 - "test":   Full Stack Trace

```

File: go.mod [hunk 2]
```diff
@@ -47,6 +47,7 @@ require (
 	github.com/onsi/gomega v1.8.1
 	github.com/opencontainers/go-digest v1.0.0-rc1
 	github.com/pkg/browser v0.0.0-20180916011732-0a3d74bf9ce4
+	github.com/pkg/errors v0.8.1
```
Comment (kdorosh) [inline go.mod]: we use eris errors now instead of github errors. they have much nicer error chaining

Comment (ilackarms) [reply go.mod]: master go.mod is out of sync, this package was introduced in #2316

Comment (ilackarms) [reply go.mod]: ill fix it anyway

**Comment by ilackarms** [comment]
/kick

```

Step #11 - "test": [2020-02-05 17:55:01.796][24882][info][main] [external/envoy/source/server/server.cc:538] main dispatch loop exited
Step #11 - "test": [AfterEach] in memory
Step #11 - "test":   /workspace/gloo/test/e2e/gateway_test.go:67
Step #11 - "test": 
Step #11 - "test": ------------------------------
Step #11 - "test": • Failure [23.035 seconds]
Step #11 - "test": Gateway in memory traffic [It] should work with no ssl and cleans up the envoy config when the virtual service is deleted 
Step #11 - "test": /workspace/gloo/test/e2e/gateway_test.go:210
Step #11 - "test": 
Step #11 - "test":   Timed out after 20.000s.
Step #11 - "test":   Expected
Step #11 - "test":       <int>: 200
Step #11 - "test":   to equal
Step #11 - "test":       <int>: 503
Step #11 - "test": 
Step #11 - "test":   /workspace/gloo/test/e2e/gateway_test.go:245
Step #11 - "test": 
Step #11 - "test":   Full Stack Trace
Step #11 - "test":   github.com/solo-io/gloo/test/e2e_test.glob..func10.1.6.4()
Step #11 - "test":   	/workspace/gloo/test/e2e/gateway_test.go:245 +0xff3
Step #11 - "test":   github.com/onsi/ginkgo/internal/leafnodes.(*runner).runSync(0xc0000ff200, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
Step #11 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/leafnodes/runner.go:113 +0xe0
Step #11 - "test":   github.com/onsi/ginkgo/internal/leafnodes.(*runner).run(0xc0000ff200, 0xc000103d68, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
Step #11 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/leafnodes/runner.go:64 +0x108
Step #11 - "test":   github.com/onsi/ginkgo/internal/leafnodes.(*ItNode).Run(0xc000905120, 0x409a680, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
Step #11 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/leafnodes/it_node.go:26 +0x88
Step #11 - "test":   github.com/onsi/ginkgo/internal/spec.(*Spec).runSample(0xc000a9d4a0, 0x0, 0x409a680, 0xc000103d40)
Step #11 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/spec/spec.go:215 +0x6e2
Step #11 - "test":   github.com/onsi/ginkgo/internal/spec.(*Spec).Run(0xc000a9d4a0, 0x409a680, 0xc000103d40)
Step #11 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/spec/spec.go:138 +0x168
Step #11 - "test":   github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).runSpec(0xc00036ef00, 0xc000a9d4a0, 0x0)
Step #11 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/specrunner/spec_runner.go:200 +0x173
Step #11 - "test":   github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).runSpecs(0xc00036ef00, 0x1)
Step #11 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/specrunner/spec_runner.go:170 +0x217
Step #11 - "test":   github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).Run(0xc00036ef00, 0x8)
Step #11 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/specrunner/spec_runner.go:66 +0x141
Step #11 - "test":   github.com/onsi/ginkgo/internal/suite.(*Suite).Run(0xc000146af0, 0x7f6e3c1ff7e8, 0xc00095bf00, 0x3a071f8, 0x9, 0xc0009910c0, 0x1, 0x1, 0x41288a0, 0xc000103d40, ...)
Step #11 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/suite/suite.go:62 +0x6a0
Step #11 - "test":   github.com/onsi/ginkgo.RunSpecsWithCustomReporters(0x409d360, 0xc00095bf00, 0x3a071f8, 0x9, 0xc0000c7ea0, 0x1, 0x1, 0xc0009910a0)
Step #11 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/ginkgo_dsl.go:226 +0x339
Step #11 - "test":   github.com/onsi/ginkgo.RunSpecs(0x409d360, 0xc00095bf00, 0x3a071f8, 0x9, 0x1)
Step #11 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/ginkgo_dsl.go:207 +0x24e
Step #11 - "test":   github.com/solo-io/gloo/test/e2e_test.TestE2e(0xc00095bf00)
Step #11 - "test":   	/workspace/gloo/test/e2e/e2e_suite_test.go:41 +0x1d6
Step #11 - "test":   testing.tRunner(0xc00095bf00, 0x3aff858)
Step #11 - "test":   	/usr/local/go/src/testing/testing.go:909 +0x19a
Step #11 - "test":   created by testing.(*T).Run
Step #11 - "test":   	/usr/local/go/src/testing/testing.go:960 +0x652
```

**Comment by ilackarms** [comment]
/kick

resource version flake

```

Step #17 - "regression-tests": â€¢ Failure [13.617 seconds]
Step #17 - "regression-tests": Kube2e: gateway tests with virtual service with a mix of valid and invalid virtual services [It] preserves the valid virtual services in envoy when a virtual service has been made invalid 
Step #17 - "regression-tests": /workspace/gloo/test/kube2e/gateway/gateway_test.go:522
Step #17 - "regression-tests": 
Step #17 - "regression-tests":   Unexpected error:
Step #17 - "regression-tests":       <*errors.withStack | 0xc000ba1b60>: {
Step #17 - "regression-tests":           error: {
Step #17 - "regression-tests":               cause: {
Step #17 - "regression-tests":                   namespace: "gateway-test-8244-1",
Step #17 - "regression-tests":                   name: "i-am-invalid",
Step #17 - "regression-tests":                   given: "",
Step #17 - "regression-tests":                   expected: "61101915",
Step #17 - "regression-tests":               },
Step #17 - "regression-tests":               msg: "reconciling resource i-am-invalid",
Step #17 - "regression-tests":           },
Step #17 - "regression-tests":           stack: [0x1d3342c, 0x1d3335c, 0x2226f67, 0x2789d1d, 0x23d5e80, 0x23d5938, 0x23d46d8, 0x23dfb52, 0x23df1b8, 0x23e72e3, 0x23e6cf7, 0x23e60e1, 0x23e9990, 0x23edc39, 0x23ed71e, 0x276f728, 0x58212a, 0x472111],
Step #17 - "regression-tests":       }
Step #17 - "regression-tests":       reconciling resource i-am-invalid: invalid resource version gateway-test-8244-1.i-am-invalid given , expected 61101915
Step #17 - "regression-tests":   occurred
Step #17 - "regression-tests": 
Step #17 - "regression-tests":   /workspace/gloo/test/kube2e/gateway/gateway_test.go:539

```

**Pull request merged**